node/npm_dependencies.ts

```diff
@@ -148,6 +148,7 @@ export const vendorNPMSpecifiers = async ({
       platform: 'node',
       plugins: [getDependencyTrackerPlugin(specifiers, importMap.getContentsWithURLObjects(), pathToFileURL(basePath))],
       write: false,
+      format: 'esm',
```

Comment:
The more general problem is that ESBuild performs all kinds of syntax checks on the code that it parses. For example, it will stumble over this snippet:

```ts
function deadCodeThatIsNeverCalled() {
  const foo = ""
  foo = "hello"
}
```

In my REPL, this didn't throw until calling it - so if it's really dead code, then it's OK to include it in the bundle. ESBuild will throw a warning, though.

This might be a problem, because it prevents code with false-positives like this from using NPM modules 🤔

Reply:
(it's not a made-up scenario, i've seen instances of this error in Humio. stuff for a follow-up though)